### PR TITLE
test: [android] run tests on the first emulator in your list

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

The objective of this pull request is to enable engineers to execute tests on emulators that are created on their local machines, instead of relying on a predefined emulator. This enhancement aims to decrease the cognitive load involved in running end-to-end (e2e) tests locally. 

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**
Smoke test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/296e5072-9e8e-479e-9807-ba833640cf2b 
Regression test run: 

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: e2e/pages/Onboarding/MetaMetricsOptInView.js [hunk 1]
```diff
@@ -27,12 +27,12 @@ class MetaMetricsOptIn {
   }
 
   async tapAgreeButton() {
-    await Gestures.swipe(this.optInMetricsContent, 'up', 'slow', 0.6);
+    await Gestures.swipe(this.optInMetricsContent, 'up', 'fast', 0.9);
```
Comment (cortisiko) [inline e2e/pages/Onboarding/MetaMetricsOptInView.js:30]: accounting for smaller screen emulators. The swipe is not long enough on some smaller to enable the "no thanks" "I agree" button

File: .detoxrc.js [hunk 2]
```diff
@@ -68,7 +77,7 @@ module.exports = {
     'android.emulator': {
       type: 'android.emulator',
       device: {
-        avdName: 'Pixel_5_API_30',
+        avdName: avdList[0],
```
Comment (cortisiko) [inline .detoxrc.js:80]: no need for predefined coded emulators anymore

File: .detoxrc.js [hunk 3]
```diff
@@ -1,4 +1,19 @@
-/** @type {Detox.DetoxConfig} */
+const { execSync } = require('child_process');
+
+const getAvailableAVDs = () => {
```
Comment (cortisiko) [inline .detoxrc.js:3]: this method basically gets the list of emulators on your local machine and run the test against the first emulator in that list

File: .detoxrc.js [hunk 4]
```diff
@@ -68,7 +76,7 @@ module.exports = {
     'android.emulator': {
       type: 'android.emulator',
       device: {
-        avdName: 'Pixel_5_API_30',
+        avdName: getAvailableAVDs[0],
```
Comment (cortisiko) [inline .detoxrc.js:79]: no more need to have predefined emulators. Tests will run on the emulator on your local machine. If you have multiple emulators, tests will run on the most recent emulator that you created

File: .detoxrc.js [hunk 5]
```diff
@@ -1,4 +1,18 @@
-/** @type {Detox.DetoxConfig} */
+const { execSync } = require('child_process');
+
+const  getAvailableAVDs = (() => {
```
Comment (cortisiko) [inline .detoxrc.js:3]: This method gets the list of emulators on your machine and puts them into an array.

File: .detoxrc.js [hunk 6]
```diff
@@ -84,11 +86,6 @@ module.exports = {
         'ios/build/Build/Products/Release-iphonesimulator/MetaMask.app',
       build: "METAMASK_BUILD_TYPE='main' METAMASK_ENVIRONMENT='production' yarn build:ios:release:e2e",
     },
-    'android.bitrise.debug': {
```
Comment (cortisiko) [inline .detoxrc.js:87]: no need for specific bitrise configs anymore. We can use the default android debug config

**Review by Cal-L — APPROVED** [review]
Lgtm!

**Pull request merged**